src/libstd/sys/redox/thread.rs

```diff
@@ -89,3 +91,18 @@ pub mod guard {
     pub unsafe fn current() -> Option<usize> { None }
     pub unsafe fn init() -> Option<usize> { None }
 }
+
+pub fn min_stack() -> usize {
```

Comment:
Instead of duplicating all this logic in all the various platforms, could just the default min stack size get implemented on various platforms?